src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -31,7 +32,13 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
             if (objectResult != null)
             {
-                SetContentTypes(objectResult.ContentTypes);
+                // Check if FormatFilter has already set the content type
```

Comment:
ah, my bad, with the newer code it will have to be ```IFormatFilter```, 
however slightly unrelated but there is another thing that we can do here , the way this is coded up makes produces dependent on IFormatFilter, even though it does nothing for the content type, instead we can do
```c# 
if (!objectResult.ConentType.Any())
{
       SetContentTypes(objectResult.ContentTypes);
}
```
This will mean that if the user has produces and is using an objectResult in his action to explicitly set the content type he is able to do so. It also means if some filter overrides/adds content type that takes precedence over Produces... 
@yishaigalatzer @rynowak .. thoughts?